base/file.jl

```diff
@@ -47,7 +47,19 @@ function mkpath(path::AbstractString, mode::Unsigned=0o777)
     dir = dirname(path)
     (path == dir || isdir(path)) && return
     mkpath(dir, mode)
-    mkdir(path, mode)
+    try
+        mkdir(path, mode)
+    # If there is a problem with making the directory, but the directory
+    # does in fact exist, then ignore the error. Else re-throw it.
+    catch err
+        # isdir comes first because I figure it will return false more often
+        # than isa
```

Comment:
I'd prefer the former. Suppressing all errors without checking the reason for them too carefully can be dangerous and conceal bugs in deeper layers.